tests/fips/gnutls/gnutls_server.pm

```diff
@@ -39,7 +39,8 @@ ca
 cert_signing_key
 EOF
     assert_script_run "certtool --generate-privkey > $pri_ca_key";
-    assert_script_run "echo '$ca_tmpl_cont' > $ca_tmpl";
+    $ca_tmpl_cont = $ca_tmpl_cont =~ s/\n/\\n/r;
```

Comment:
This line should be `$ca_tmpl_cont =~ s/\n/\\n/g;`

The `/r` is not needed, the `/g` is important however, otherwise the regex will replace only one newline. Same applies to the other changes.

Reply:
Typo, of course! It should have been a `/g`, don't ask me how it became a `/r`.

Reply:
Fixed.

Reply:
Not done yet, now you're replacing the string with boolean value whether the regex matched something.

Reply:
:facepalm: